client/keys/utils.go

```diff
@@ -29,6 +29,54 @@ func GetKeyBase() (keys.Keybase, error) {
 	return GetKeyBaseFromDir(rootDir)
 }
 
+// GetKeyInfo returns key info for a given name. An error is returned if the
+// keybase cannot be retrieved or getting the info fails.
+func GetKeyInfo(name string) (keys.Info, error) {
+	keybase, err := GetKeyBase()
+	if err != nil {
+		return nil, err
+	}
+
+	return keybase.Get(name)
+}
+
+// GetPassphrase returns a passphrase for a given name. It will first retrieve
+// the key info for that name if the type is local, it'll fetch input from
+// STDIN. Otherwise, an empty passphrase is returned. An error is returned if
+// the key info cannot be fetched or reading from STDIN fails.
+func GetPassphrase(name string) (string, error) {
+	var passphrase string
+
+	keyInfo, err := GetKeyInfo(name)
+	if err != nil {
+		return passphrase, err
+	}
+
+	// we only need a passphrase for locally stored keys
+	if keyInfo.GetType() == keys.TypeLocal {
+		passphrase, err = ReadPassphraseFromStdin(name)
```

Comment:
TODO: Make Issue to address this before launch

Reply:
(we shouldn't be passing around passphrases even in memory)

Reply:
There already is an issue for this. #864

Reply:
How would we avoid storing the passphrase (transiently) in memory? We need to decrypt the encrypted keyfile.

Reply:
@cwgoes brings up a good point. Warrants further discussion. I'll add a reference to #864 in the code tho